File: AndroidRoom/app/src/main/java/android/room/view/ui/NewWordActivity.java
package android.room.view.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.room.view.databinding.ActivityNewWordBinding;
import android.text.TextUtils;

public class NewWordActivity extends AppCompatActivity {

    ActivityNewWordBinding binding;
    static final String EXTRA_REPLY = "android.room.view";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        binding = ActivityNewWordBinding.inflate(getLayoutInflater());
        super.onCreate(savedInstanceState);
        setContentView(binding.getRoot());

        String mEditWordView = binding.editWord.getText().toString();

        binding.buttonSave.setOnClickListener(view -> {
            Intent replyIntent = new Intent();
            if (TextUtils.isEmpty(mEditWordView)){
                setResult(RESULT_CANCELED, replyIntent);
            } else {
                replyIntent.putExtra(EXTRA_REPLY, mEditWordView);
                setResult(RESULT_OK, replyIntent);
            }
            finish();
        });
    }
}

File: RoomWordSample/app/src/main/java/android/codelab/roomwordsample/NewWordActivity.kt
package android.codelab.roomwordsample

import android.app.Activity
import android.codelab.roomwordsample.databinding.ActivityNewWordBinding
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils

class NewWordActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNewWordBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        binding = ActivityNewWordBinding.inflate(layoutInflater)
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.buttonSave.setOnClickListener {
            val replyIntent = Intent()
            if (TextUtils.isEmpty(binding.editWord.text)) {
                setResult(Activity.RESULT_CANCELED, replyIntent)
            } else {
                val word = binding.editWord.text.toString()
                replyIntent.putExtra(EXTRA_REPLY, word)
                setResult(Activity.RESULT_OK, replyIntent)
            }
            finish()
        }
    }

    companion object {
        const val EXTRA_REPLY = "com.example.android.wordlistsql.REPLY"
    }
}